src/Kernel-CodeModel/CompiledMethod.class.st

```diff
@@ -925,9 +913,16 @@ CompiledMethod >> sourcePointer [
 { #category : 'source code management' }
 CompiledMethod >> sourcePointer: srcPointer [
 
+	| trailerBytes trailerSize start |
 	"Drop the #source property if any"
 	self removeProperty: #source.
-	self setSourcePointer: srcPointer
+	trailerSize := self class trailerSize.
+	trailerBytes := srcPointer asByteArrayOfSize: trailerSize.
+	start := self size - trailerSize.
+	start + 1 to: self size do: [ :n | self at: n put: (trailerBytes at: n - start) ].
+
+	"In case the mtehod is definied in a trait, we need to update the source pointers of the copies."
+	self origin = self methodClass ifTrue: [ self origin notifySourcePointerOf: self selector updateTo: self sourcePointer ]
```

Comment:
Hmm... updating the source pointer of the copies feels wrong. If the original will be changes, it will be re-installed anyway, or not?